force-app/main/default/classes/SOQL.cls

```diff
@@ -130,9 +157,10 @@ public with sharing class SOQL {
      * @return A list of Lead records.
      * Update the corresponding test class and testGetLeadsBySourceOrTitle method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
+    public static List<Lead> getLeadsBySourceOrTitle (String paramLeadSource, String paramTitle) {
     //  The body of the method
-    //}    
+        return Database.query('SELECT Id, FirstName, LastName, Title, Company, Phone, Email, LeadSource, Status FROM Lead where LeadSource like \'%' + paramLeadSource + '%\'' + ' OR Title like \'%' + paramTitle + '%\'');
```

Comment:
Generally speaking, I would not make a query as a string or dynamically unless I needed to. In this case, I love to see your experimenting with it. But, the benefit of writing a query statically is that the compiler checks to make sure it is a valid query before you run it. It can't do that when you build a query with a String.